velox/functions/sparksql/SplitFunctions.cpp

```diff
@@ -134,10 +135,131 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    DecodedVector* inputString = decodedArgs.at(0);
+
+    BaseVector::ensureWritable(
+        rows, MAP(VARCHAR(), VARCHAR()), context.pool(), result);
+    exec::VectorWriter<Map<Varchar, Varchar>> resultWriter;
+    resultWriter.init(*result->as<MapVector>());
+
+    folly::F14FastMap<StringView, vector_size_t> keyToIdx;
+    rows.applyToSelected([&](vector_size_t row) {
+      resultWriter.setOffset(row);
+      auto& mapWriter = resultWriter.current();
+
+      const StringView& current = inputString->valueAt<StringView>(row);
+      const char* pos = current.begin();
+      const char* end = pos + current.size();
+      const char* entryEnd;
+      const char* keyEnd;
+      do {
+        entryEnd = std::find(pos, end, entryDelim_);
+        keyEnd = std::find(pos, entryEnd, keyValueDelim_);
+        auto key = StringView(pos, keyEnd - pos);
+        auto iter = keyToIdx.find(key);
+        VELOX_USER_CHECK(
+            iter == keyToIdx.end(),
+            "Duplicated keys ('{}') are not allowed.",
+            key);
+        keyToIdx.emplace(key, mapWriter.size());
+        if (keyEnd == entryEnd) {
+          mapWriter.add_null().append(key);
+        } else {
+          auto [keyWriter, valueWriter] = mapWriter.add_item();
+          keyWriter.setNoCopy(key);
+          valueWriter.setNoCopy(StringView(keyEnd + 1, entryEnd - keyEnd - 1));
+        }
+        pos = entryEnd + 1; // Skip past delim.
+      } while (entryEnd != end);
+
+      resultWriter.commit();
+    });
+
+    resultWriter.finish();
+
+    // Reuses input buffer.
+    result->as<MapVector>()
+        ->mapKeys()
+        ->as<FlatVector<StringView>>()
+        ->acquireSharedStringBuffers(inputString->base());
+    result->as<MapVector>()
+        ->mapValues()
+        ->as<FlatVector<StringView>>()
+        ->acquireSharedStringBuffers(inputString->base());
+  }
+
+ private:
+  char entryDelim_;
+  char keyValueDelim_;
+};
+
+/// Currently only supports single-character entryDelim & keyValueDelim and
+/// these two delimiters must be constant.
+std::shared_ptr<exec::VectorFunction> createStringToMap(
+    const std::string& /*name*/,
+    const std::vector<exec::VectorFunctionArg>& inputArgs,
+    const core::QueryConfig& /*config*/) {
+  char entryDelim;
+  char keyValueDelim;
+
+  VELOX_USER_CHECK(
+      inputArgs.size() == 3, "Expected 3 args for StringToMap function.");
+  BaseVector* constantVector = inputArgs[1].constantValue.get();
+  VELOX_USER_CHECK(
+      constantVector != nullptr,
+      "StringToMap function only allows constant entry delimiter.");
+  auto constantStringView =
+      constantVector->as<ConstantVector<StringView>>()->valueAt(0);
```

Comment:
Please ensure the constant is not null. Same for the inputArgs[2].

Reply:
Thanks for you review! Fixed.